applications/ExaquteSandboxApplication/custom_processes/metrics_divergencefree_process.cpp

```diff
@@ -0,0 +1,385 @@
+            mMeanStrategyDivergenceFreeOverAllDomain += std::pow(it_elem->GetValue(r_reference_var),2);
+        }
+        mMeanStrategyDivergenceFreeOverAllDomain = std::sqrt(mMeanStrategyDivergenceFreeOverAllDomain);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TDim>
+void MetricDivergenceFreeProcess<TDim>::CalculateMetric()
+{
+    // Prepare arrays
+    NodesArrayType& nodes_array = mrThisModelPart.Nodes();
+    ElementsArrayType& elements_array = mrThisModelPart.Elements();
+
+    // Tensor variable definition
+    const Variable<TensorArrayType>& tensor_variable = KratosComponents<Variable<TensorArrayType>>::Get("METRIC_TENSOR_"+std::to_string(TDim)+"D");
+
+    // Prepare iterations over all nodes and elements
+    const int number_nodes = static_cast<int>(nodes_array.size());
+    const int number_elements = static_cast<int>(elements_array.size());
+    KRATOS_DEBUG_ERROR_IF(number_nodes == 0) <<  "ERROR:: Empty list of nodes" << std::endl;
+
+    const auto& r_reference_var  = KratosComponents<Variable<double>>::Get(mReferenceVariable);
+
+    //  Compute nodal divergence loops
+    Vector divergence_nodal_values(number_nodes);
+    Vector area_nodal_values(number_nodes);
+    // Initialize to zero
+    for (int i_node = 0; i_node < number_nodes; ++i_node) {
+        divergence_nodal_values[i_node] = 0;
+        area_nodal_values[i_node] = 0;
+    }
+    // Weighted mean
+    const auto it_elem_begin = elements_array.begin();
+    for (int i_elem = 0; i_elem < number_elements; ++i_elem) {
+        auto it_elem = it_elem_begin + i_elem;
+        auto r_geometry = it_elem->GetGeometry();
+        for (unsigned int i_node = 0; i_node < r_geometry.size(); ++i_node) {
+            area_nodal_values[r_geometry[i_node].Id()-1] = area_nodal_values[r_geometry[i_node].Id()-1] + it_elem->GetGeometry().Area() / r_geometry.size();
+            divergence_nodal_values[r_geometry[i_node].Id()-1] = divergence_nodal_values[r_geometry[i_node].Id()-1] + it_elem->GetValue(r_reference_var) * it_elem->GetGeometry().Area() / r_geometry.size();
+        }
+    }
+    // Delete area contribution
+    for (int i_node = 0; i_node < number_nodes; ++i_node) {
+        divergence_nodal_values[i_node] = divergence_nodal_values[i_node] / area_nodal_values[i_node];
+    }
+
+    // Maximum strategy
+    if (mRefinementStrategy == MaximumStrategy) {
+
+        const auto it_nodes_begin = nodes_array.begin();
+        #pragma omp parallel for
+        for(int i_node = 0; i_node < number_nodes; ++i_node) {
+            auto it_node = it_nodes_begin + i_node;
+
+            // Retrieve variable
+            const double divergencefree_elem_value = divergence_nodal_values[i_node];
+
+            // Estimate element size
+            const double nodal_h = it_node->GetValue(NODAL_H);
```

Comment:
You should add a check at least in debug, if some variable is not defined GetValue returns 0.0 for a double

Reply:
Done